src/main/java/org/codehaus/mojo/buildhelper/AddResourceMojo.java

```diff
@@ -37,10 +38,22 @@
 @Mojo(name = "add-resource", defaultPhase = LifecyclePhase.GENERATE_RESOURCES, threadSafe = true)
 public class AddResourceMojo extends AbstractAddResourceMojo {
 
+    /**
+     * If a resource directory does not exist, do not add it as a root.
+     *
+     * @since 3.5.0
+     */
+    @Parameter(property = "buildhelper.addresource.skipIfMissing", defaultValue = "false")
+    private boolean skipIfMissing;
```

Comment:
Similar here we have dedicated property by goals, so will be appreciated to have also dedicated configuration item.